{{cookiecutter.github_repository_name}}/src/{{cookiecutter.module_name}}/serializers.py

```diff
@@ -30,55 +37,65 @@ def validate(self, validated_data):
         except APIException:
             raise serializers.ValidationError({"token": ["Invalid company."]})
 
-        if Company.objects.filter(identifier=company['identifier']).exists():
+        if Company.objects.filter(
+                identifier=company['id'],
+                active=True).exists():
             raise serializers.ValidationError(
                 {"token": ["Company already activated."]})
 
-        try:
-            currencies = rehive.company.currencies.get()
-        except APIException:
-            raise serializers.ValidationError({"non_field_errors":
-                ["Unkown error."]})
-
         validated_data['user'] = user
         validated_data['company'] = company
-        validated_data['currencies'] = currencies
 
         return validated_data
 
+    @transaction.atomic
     def create(self, validated_data):
         token = validated_data.get('token')
         rehive_user = validated_data.get('user')
         rehive_company = validated_data.get('company')
-        currencies = validated_data.get('currencies')
-
-        with transaction.atomic():
-            user = User.objects.create(token=token,
-                identifier=uuid.UUID(rehive_user['identifier']).hex)
-
-            company = Company.objects.create(admin=user,
-                identifier=rehive_company.get('identifier'),
-                name=rehive_company.get('name'))
 
+        # Activate an existing company.
+        try:
+            company = Company.objects.get(
+                identifier=rehive_company.get('id')
+            )
+        # Ceate a new company and activate it.
+        except Company.DoesNotExist:
+            with transaction.atomic():
+                user = User.objects.create(
+                    token=token,
+                    identifier=uuid.UUID(rehive_user['id'])
+                )
+                company = Company.objects.create(
+                    admin=user,
+                    identifier=rehive_company.get('id'),
+                    email=rehive_company.get('email'),
```

Comment:
You can remove this `email` line. As far as I know this is only needed for the notification service.